packages/go_router/lib/src/route_data.dart

```diff
@@ -82,8 +86,14 @@ abstract class GoRouteData {
     Widget builder(BuildContext context, GoRouterState state) =>
         factoryImpl(state).build(context);
 
-    Page<void> pageBuilder(BuildContext context, GoRouterState state) =>
-        factoryImpl(state).buildPage(context);
+    Page<void> pageBuilder(BuildContext context, GoRouterState state) {
+      final T routeData = factoryImpl(state);
+      final Page<void> page = routeData.buildPageWithState(context, state);
+      if (page is! NoOpPage) {
```

Comment:
you can skip this check and just return the page if you make buildPageWithState return buildPage by default

Reply:
👍 I've applied your suggestion in https://github.com/flutter/packages/pull/2444/commits/b67e5f8fa369ad12631b072f325416a5881136bc